app/views/foreman_discovery/debian_kexec.erb

```diff
@@ -29,12 +32,31 @@ Extra options like --reset-vga can be set via "extra" array.
   options << @host.facts['append']
   options << "domain=#{@host.domain}"
   options << 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true'
-  options << "locale=#{host_param('lang') || 'en_US'}"
+  options << "locale=#{@host.params['lang'] || 'en_US'}"
   options << "inst.stage2=#{@host.operatingsystem.medium_uri(@host)}" if @host.operatingsystem.name.match(/Atomic/i)
+  
+  if subnet4 && !subnet4.dhcp_boot_mode?
+    foreman_url = foreman_url('provision', static: '1')
+    options << "netcfg/disable_dhcp=true"
+    options << "netcfg/get_ipaddress=#{ip}"
+    options << "netcfg/get_netmask=#{mask}"
+    options << "netcfg/get_gateway=#{gw}"
+    options << "netcfg/get_nameservers=#{dns}"
+  elsif subnet6 && !subnet6.dhcp_boot_mode?
+    foreman_url = foreman_url('provision', static6: '1')
+    options << "netcfg/disable_dhcp=true"
+    options << "netcfg/get_ipaddress=#{ip}"
+    options << "netcfg/get_netmask=#{mask}"
+    options << "netcfg/get_gateway=#{gw}"
+    options << "netcfg/get_nameservers=#{dns}"
+  else
+    foreman_url = foreman_url('provision')
+    options << "netcfg/disable_dhcp=false"
```

Comment:
Well I am not sure if this is correct. It looks like the installer will not recognize these options, only disable_dhcp according to docs:

https://www.debian.org/releases/jessie/amd64/ch05s03.html.en

These are preseed options:

https://www.debian.org/releases/stretch/amd64/apbs04.html.en#preseed-network

Have you tested this?